### DISCUSSION TITLE
About authentication system and token generation in Hubs

### DISCUSSION THREAD

**Discussion by jasoncks** [mozilla/hubs#3472] [Hubs Cloud Questions] [answered]
I want to integrate another authentication system to hubs, such that user can login using there social account.

In this case, most likely I want skip the "email verification" part in hubs.
Is this part can be skip? Which scripts should I modify?

Also, as I know, when doing API request, we need to have token generated from "Reticulum" server, how do this token generated and how we can obtain this Token?

**Comment by robinkwilson** [comment, marked as answer]
Auth integration with other systems are being looked at right now: https://github.com/mozilla/hubs/issues/2312

Token is located in localstorage on the client.
Currently there's no way to modify the backend reticulum code. So any modifications to the flow would need to happen on the client, then you deploy a custom client to your hubs-cloud instance: https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html

**Reply by robinkwilson** [reply]
Marking as answered to remove thread from our pipeline, if needed feel free to "Unmark as answer".
